web/src/components/software/PatternSelector.jsx

```diff
@@ -0,0 +1,178 @@
+    if (pattern_groups[pattern.group]) {
+      pattern_groups[pattern.group].push(pattern);
+    } else {
+      pattern_groups[pattern.group] = [pattern];
+    }
+  });
+
+  // sort patterns by the "order" value
+  Object.keys(pattern_groups).forEach((group) => {
+    pattern_groups[group].sort((p1, p2) => p1.order === p2.order ? (p1.name < p2.name) : (p1.order < p2.order ? -1 : 1));
+  });
+
+  return pattern_groups;
+}
+
+function sortGroups(groups) {
+  return Object.keys(groups).sort((g1, g2) => {
+    const order1 = groups[g1][0].order;
+    const order2 = groups[g2][0].order;
+
+    if (order1 === order2) {
+      return g1 === g2 ? 0 : (g1 < g2 ? -1 : 1);
+    }
+
+    return order1 < order2 ? -1 : 1;
+  });
+}
+
+function PatternSelector() {
+  const [patterns, setPatterns] = useState();
+  const [selected, setSelected] = useState();
+  const [used, setUsed] = useState();
+  const [searchValue, setSearchValue] = useState("");
+  const client = useInstallerClient();
+
+  const onSearchChange = (value) => {
+    setSearchValue(value);
+  };
+
+  // refresh the page content after changing a pattern status
+  const refreshCb = useCallback(() => {
+    const refresh = async () => {
+      setSelected(await client.software.selectedPatterns());
+      setUsed(await client.software.getUsedSpace());
+    };
+
+    refresh();
+  }, [client.software]);
+
+  useEffect(() => {
+    // patterns already loaded
+    if (patterns) return;
+
+    const loadData = async () => {
+      setPatterns(await client.software.patterns(true));
+      setSelected(await client.software.selectedPatterns());
+      setUsed(await client.software.getUsedSpace());
+    };
+
+    loadData();
+  }, [patterns, client.software]);
+
+  if (!patterns || !selected) {
+    return <></>;
+  }
+
+  let patternsData = convert(patterns, selected);
+
+  // count the number of selected patterns
+  const numSelected = patternsData.reduce((acc, pat) => acc + (pat.selected === undefined ? 0 : 1), 0);
+
+  // filtering - search the required text in the name and pattern description
+  if (searchValue !== "") {
+    const searchData = searchValue.toUpperCase();
+    patternsData = patternsData.filter((p) =>
+      p.name.toUpperCase().indexOf(searchData) !== -1 ||
+      p.description.toUpperCase().indexOf(searchData) !== -1
+    );
+  }
+
+  const groups = groupPatterns(patternsData);
+  if (process.env.NODE_ENV !== "production") console.log("patterns: ", groups);
+
+  const sortedGroups = sortGroups(groups);
+  if (process.env.NODE_ENV !== "production") console.log("sorted groups: ", sortedGroups);
+
+  const selector = sortGroups(groups).map((group) => {
+    return (
+      <PatternGroup
+        key={group}
+        name={group}
+      >
+        { (groups[group]).map(p => <PatternItem key={p.name} pattern={p} onChange={refreshCb} />) }
+      </PatternGroup>
+    );
+  });
+
+  return (
+    <>
+      <Section title={_("Selected Software")} icon="apps">
```

Comment:
> For me, this section could be rendered without title because, after looking at the screenshot and reading comments from @joseivanlopez, @shundhammer, and yourself I see it as an _information area_ of the page. A section (without title) for wrapping the page tools (the filter) and the basic page information (the product name - since we're in the Software page -, the size it will take, the number of patterns/packages selected...

Some thoughts about the product name:

a) I am not sure whether showing it in that "information are" of the Software page is a good idea. Definitely it is something you cannot change in this page.
b) I also miss the information about the selected product in other pages (storage, users, etc). Perphaps somehow we should have the product name always visible.

Reply:
Of course, something out of the scope of this PR